pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java

```diff
@@ -560,4 +562,42 @@ public static String decodeUrl(String input)
       throws UnsupportedEncodingException {
     return URLDecoder.decode(input, StandardCharsets.UTF_8.toString());
   }
+
+  /**
+   * @param input binary data
+   * @return Base64 encoded String
+   */
+  @ScalarFunction
+  public static String binaryToBase64(byte[] input) {
+    return Base64.getEncoder().encodeToString(input);
```

Comment:
What is the expectation on charset of input bytes / binary data ? Looks like we assume it is UTF8. What if the input bytes provided here was encoded using UTF16

Reply:
Resolved - we will not have a fixed charset of input bytes now with the new implementation. For example, if user wants to encode the String "hello!", he/she can choose to do `toBase64(toUtf8("hello"))` or `toBase64(toUtf16("hello"))` (Pinot currently only supports toUtf8 and toASCII functions). During decoding, he/she can do `fromUtf8(fromBase64("fvduivheui"))` to `fromUtf16(fromBase64("fvduivheui"))` to get "hello!" depending on if they used UTF8 or UTF16 when they encoded.

Reply:
Resolved - added tests for calling toBase64 on a string. Will add in documentation that BYTES col will be represented in HEX string(also link:https://docs.pinot.apache.org/users/user-guide-query/querying-pinot#bytes-column).